CreeDictionary/utils/crkeng_xml_utils.py

```diff
@@ -79,3 +106,237 @@ def convert_xml_inflectional_category_to_word_class(
         return WordClass.IPV
 
     return None
+
+
+class IndexedXML(Iterable[XMLEntry]):
+    """
+    The collection of all entries parsed from the XML source file.
+    It supports fast querying with field/field combinations by hashing the entries in advance.
+    """
+
+    # --- alias for type hints---
+    Indexes = Dict[
+        Tuple[NamedTupleFieldName, ...],
+        Dict[Tuple[NamedTupleFieldValue, ...], Set[XMLEntry]],
+    ]
+
+    # --- Instance attribute type hints ---
+    _entries: Set[XMLEntry]
+    _indexes: Indexes
+    _source_abbreviations: Iterable[str]
+
+    # --- Class attributes ---
+    # Add key or key combinations here when it's needed to lookup with certain keys
+    # order does not matter here, they will be sorted in the built index
+    INDEX_KEYS: List[Tuple[NamedTupleFieldName, ...]] = [
+        # (
+        #     NamedTupleFieldName("l"),
+        #     NamedTupleFieldName("pos"),
+        #     NamedTupleFieldName("ic"),
+        # ),
+        (NamedTupleFieldName("l"),),
+    ]
```

Comment:
@eddieantonio this is worth your consideration!!
soooo....I started this PR with this `IndexedXML` class. At that time, I thought I'd need to query the xml in different ways, like 
`indexed_xml.filter(l="niska")`
and
`indexed_xml.filter(l="niska", pos="N", ic="NA")`

Thus I came up with this solution of customizable combination of fields to hash indexes ahead of time.

It later turned out I only needed to have `filter(l="xxx")` in the importer code.

so I commented out the first "index" here.

It doesn't affact any performace. But changing the code here to only consider the `filter(l="xxx")` case will improve the readibility. 

On the other hand, 1. I don't want to delete this amazing code I wrote. 2. This might be a viable solution to JIT feature we will have later, as we'll definitly need to query the xml in different ways.

Tell me if you think this is a technical debt and whether I should refactor this!